post_processing/ci_plots.jl

```diff
@@ -542,19 +543,112 @@ EDMFBoxPlots = Union{
-    )
+
+    summary_files = make_plots_edmf_vars_z(simulation_path, vars_z)
+    make_plots_generic(simulation_path, vars_zt, summary_files = summary_files)
+end
+
+long_name(var) = var.attributes["long_name"]
+short_name(var) = var.attributes["short_name"]
+
+function parse_long_name(var)
+    name = long_name(var)
+    for str in ["east ", "west ", "north ", "south "]
+        name = replace(name, "degrees_" * str => "")
+    end
+    var_name, rest = split(name, ",")
+    reduction, rest = split(rest, "(s) ")
+    coord, time = split(rest, " time = ")
+    return "$(short_name(var)), $coord, t = $time"
+end
+
+function make_plots_edmf_vars_z(output_path, vars_z)
+    # Group gridmean and updraft variables
+    get_longname_suffix(v) =
+        startswith(long_name(v), "Updraft") ? long_name(v)[9:end] : long_name(v)
+    # OrderedDict will keep the same variables at different lat/lon together
+    grouped_vars = OrderedDict{String, Vector{ClimaAnalysis.OutputVar}}()
+    for var in sort(vars_z, by = get_longname_suffix)
+        suffix = get_longname_suffix(var)
+        if haskey(grouped_vars, suffix)
+            push!(grouped_vars[suffix], var)
+        else
+            grouped_vars[suffix] = [var]
+        end
+    end
+
+    # Setup plotting and layout variables
+    z = vars_z[1].dims["z"]
+
+    MAX_ROWS_PER_PAGE = 4
+    MAX_COLS_PER_ROW = 2
+    MAX_PLOTS_PER_PAGE = MAX_ROWS_PER_PAGE * MAX_COLS_PER_ROW
+
+    makefig() = CairoMakie.Figure(resolution = (900, 300 * MAX_ROWS_PER_PAGE))
+    fig = makefig()
+    page = 1
+    summary_files = String[]
+    gridlayout = []
+    for (var_num, var_group) in enumerate(values(grouped_vars))
+        if var_num % MAX_PLOTS_PER_PAGE == 1
+            fig = makefig()
+            gridlayout = map(1:MAX_PLOTS_PER_PAGE) do i
+                row = div(i - 1, MAX_COLS_PER_ROW) % MAX_ROWS_PER_PAGE + 1
+                col = (i - 1) % MAX_COLS_PER_ROW + 1
+                fig[row, col] = CairoMakie.GridLayout()
+            end
+        end
+
+        units = var_group[1].attributes["units"]
+        ax = CairoMakie.Axis(
+            gridlayout[(var_num - 1) % MAX_PLOTS_PER_PAGE + 1][1, 1],
+            ylabel = "z [$(vars_z[1].dim_attributes["z"]["units"])]",
+            xlabel = "$(short_name(var_group[1])) [$units]",
+            title = parse_long_name(var_group[1]),
+            titlesize = 14,
+        )
+        for var in var_group
+            CairoMakie.lines!(ax, var.data, z, label = short_name(var))
+        end
+        # Create a legend for gridmean/updraft pairs
+        length(var_group) > 1 && Makie.axislegend(ax)
+        # Flush current page
+        if var_num % MAX_PLOTS_PER_PAGE == 0 ||
+           var_num == length(keys(grouped_vars))
+            file_path = joinpath(output_path, "edmf_vars_z_temp_$page.pdf")
+            CairoMakie.save(file_path, fig)
+            push!(summary_files, file_path)
+            page += 1
+        end
+    end
+    return summary_files
```

Comment:
A lot of the logic here is for page handling. I think it would be nice if they were abstracted away and added to `make_plots_generic`. Then `make_plot_generic` could take a plotting function.